receiver/postgresqlreceiver/config_test.go

```diff
@@ -0,0 +1,139 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package postgresqlreceiver
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/config/configtls"
+	"go.uber.org/multierr"
+)
+
+func TestValidate(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		cfg      *Config
+		expected error
+	}{
+		{
+			desc: "missing username and password",
+			cfg:  &Config{},
```

Comment:
Something we should think about: Is it more valuable to test configs that are built up precisely like this, or should we start with the default config and mutate it. The latter would directly reflect the way a user actually configures the receiver, and would be another check against changes to the default config. This would be a good point of standardization. What do you think about the options?

Reply:
I think this makes sense, changed the way the config tests worked so they modified a default config rather than providing their own config to each test case.